### ISSUE TITLE
Not able to list all images, Image.find() only work when pass product_id

### ISSUE THREAD

**Issue by nsuberi** [Shopify/shopify_python_api#383]
I can successfully retrieve a list of all products, and retrieve images linked to a specific product. I try to get a list of all images that I can use to link my product listings with the CDN urls shopify generates when I upload images. I try to use the code below and get a 404 that the resource doesn't exist. If I go directly to the url in question in my browser it works.

`shopify.Image.find(product_id=None)`

Url that works but throws an error in this library: https://enchanted-art-sole-comfort.myshopify.com/admin/images.json

Any help appreciated!

~ Nathan
 
```
single_image = shopify.Image.find(product_id=products[0].id)
print(single_image)
```
`[[image(14136140038210)]]`

# Doesn't work: 

`all_images = shopify.Image.find(product_id=None)`

```
---------------------------------------------------------------------------
HTTPError                                 Traceback (most recent call last)
~/.pyenv/versions/3.6.0/lib/python3.6/site-packages/pyactiveresource/connection.py in _open(self, method, path, headers, data)
    285             try:
--> 286                 http_response = self._handle_error(self._urlopen(request))
    287             except urllib.error.HTTPError as err:

~/.pyenv/versions/3.6.0/lib/python3.6/site-packages/pyactiveresource/connection.py in _urlopen(self, request)
    315         if _urllib_has_timeout():
--> 316           return urllib.request.urlopen(request, timeout=self.timeout)
    317         else:

~/.pyenv/versions/3.6.0/lib/python3.6/urllib/request.py in urlopen(url, data, timeout, cafile, capath, cadefault, context)
    222         opener = _opener
--> 223     return opener.open(url, data, timeout)
    224 

~/.pyenv/versions/3.6.0/lib/python3.6/urllib/request.py in open(self, fullurl, data, timeout)
    531             meth = getattr(processor, meth_name)
--> 532             response = meth(req, response)
    533 

~/.pyenv/versions/3.6.0/lib/python3.6/urllib/request.py in http_response(self, request, response)
    641             response = self.parent.error(
--> 642                 'http', request, response, code, msg, hdrs)
    643 

~/.pyenv/versions/3.6.0/lib/python3.6/urllib/request.py in error(self, proto, *args)
    569             args = (dict, 'default', 'http_error_default') + orig_args
--> 570             return self._call_chain(*args)
    571 

~/.pyenv/versions/3.6.0/lib/python3.6/urllib/request.py in _call_chain(self, chain, kind, meth_name, *args)
    503             func = getattr(handler, meth_name)
--> 504             result = func(*args)
    505             if result is not None:

~/.pyenv/versions/3.6.0/lib/python3.6/urllib/request.py in http_error_default(self, req, fp, code, msg, hdrs)
    649     def http_error_default(self, req, fp, code, msg, hdrs):
--> 650         raise HTTPError(req.full_url, code, msg, hdrs, fp)
    651 

HTTPError: HTTP Error 404: Not Found

During handling of the above exception, another exception occurred:

ResourceNotFound                          Traceback (most recent call last)
<ipython-input-49-4b22cca1d13c> in <module>()
      2 print(single_image)
      3 # Doesn't work:
----> 4 all_images = shopify.Image.find(product_id=None)

~/.pyenv/versions/3.6.0/lib/python3.6/site-packages/shopify/base.py in find(cls, id_, from_, **kwargs)
    209     def find(cls, id_=None, from_=None, **kwargs):
    210         """Checks the resulting collection for pagination metadata."""
--> 211         collection = super(ShopifyResource, cls).find(id_=id_, from_=from_, **kwargs)
    212         if isinstance(collection, Collection) and "headers" in collection.metadata:
    213             return PaginatedCollection(collection, metadata={"resource_class": cls}, **kwargs)

~/.pyenv/versions/3.6.0/lib/python3.6/site-packages/pyactiveresource/activeresource.py in find(cls, id_, from_, **kwargs)
    384             return cls._find_single(id_, **kwargs)
    385 
--> 386         return cls._find_every(from_=from_, **kwargs)
    387 
    388     @classmethod

~/.pyenv/versions/3.6.0/lib/python3.6/site-packages/pyactiveresource/activeresource.py in _find_every(cls, from_, **kwargs)
    523             path = cls._collection_path(prefix_options, query_options)
    524 
--> 525         response = cls.connection.get(path, cls.headers)
    526         objs = cls.format.decode(response.body)
    527         return cls._build_collection(objs, prefix_options, response.headers)

~/.pyenv/versions/3.6.0/lib/python3.6/site-packages/pyactiveresource/connection.py in get(self, path, headers)
    327             A Response object.
    328         """
--> 329         return self._open('GET', path, headers=headers)
    330 
    331     def get_formatted(self, path, headers=None):

~/.pyenv/versions/3.6.0/lib/python3.6/site-packages/shopify/base.py in _open(self, *args, **kwargs)
     25         self.response = None
     26         try:
---> 27             self.response = super(ShopifyConnection, self)._open(*args, **kwargs)
     28         except pyactiveresource.connection.ConnectionError as err:
     29             self.response = err.response

~/.pyenv/versions/3.6.0/lib/python3.6/site-packages/pyactiveresource/connection.py in _open(self, method, path, headers, data)
    286                 http_response = self._handle_error(self._urlopen(request))
    287             except urllib.error.HTTPError as err:
--> 288                 http_response = self._handle_error(err)
    289             except urllib.error.URLError as err:
    290                 raise Error(err, url)

~/.pyenv/versions/3.6.0/lib/python3.6/site-packages/pyactiveresource/connection.py in _handle_error(self, err)
    417             raise ForbiddenAccess(err)
    418         elif err.code == 404:
--> 419             raise ResourceNotFound(err)
    420         elif err.code == 405:
    421             raise MethodNotAllowed(err)

ResourceNotFound: Not Found: https://enchanted-art-sole-comfort.myshopify.com/admin/images.json
```

**Comment by tanema** [comment]
This is weird, I tested it out and it seems to raise 404 even though the URL is correct. I am not sure why that would be. I will have to look into this closer soon. I thought it might be permissions but I made sure that was not the issue so I am not sure why this would happen or why it would return 404

**Comment by nsuberi** [comment]
Thank you sir! I set all my permissions to read. Doesn’t seem like there is
an option specifically addressing file/image access in the shopify private
app manager.

I notice in the tests that there is no case for retrieving all images, only
ones specifically link to a product.

I made a workaround by copying the content of that url into my notebook and
then searching from there.

I can link products by creating an image with
Shopify.Image({‘product_id’:_id}) and then assigning the src to be the
public url from what I copied through the workarounds and then adding that
to a product.image and product.images attributes.

~ Nathan
